src/v2i-hub/PedestrianPlugin/src/PedestrianPlugin.cpp

```diff
@@ -226,37 +247,104 @@ void PedestrianPlugin::OnStateChange(IvpPluginState state)
-
-	auto msg = std::make_unique<PsmEncodedMessage>();
-	msg.reset();
-	msg.reset(dynamic_cast<PsmEncodedMessage*>(factory.NewMessage(api::MSGSUBTYPE_PERSONALSAFETYMESSAGE_STRING)));
 
-	std::string enc = psmENC.get_encoding();
-	msg->refresh_timestamp();
-	msg->set_payload(psmENC.get_payload_str());
-	msg->set_encoding(enc);
-	msg->set_flags(IvpMsgFlags_RouteDSRC);
-	msg->addDsrcMetadata(tmx::messages::api::personalSafetyMessage_PSID);
-	msg->refresh_timestamp();
+	PLOG(logDEBUG) << "In PedestrianPlugin::BroadcastPedDet: Incoming Message: " << std::endl << msgJson;
+	auto receivedType = parseMessage(msgJson);
 
-	auto *rMsg = dynamic_cast<routeable_message *>(msg.get());
-
-	BroadcastMessage(*rMsg);
-
-	PLOG(logINFO) << " Pedestrian Plugin :: Broadcast PSM:: " << psmENC.get_payload_str();
+	if(receivedType == "PersonalSafetyMessage")
+	{
+		psmMsg.set_contents(container.get_storage().get_tree());
+		psmENC.encode_j2735_message(psmMsg);
+
+		auto msg = std::make_unique<PsmEncodedMessage>();
+		msg.reset();
+		msg.reset(dynamic_cast<PsmEncodedMessage*>(factory.NewMessage(api::MSGSUBTYPE_PERSONALSAFETYMESSAGE_STRING)));
+		std::string enc = psmENC.get_encoding();
+		msg->refresh_timestamp();
+		msg->set_payload(psmENC.get_payload_str());
+		msg->set_encoding(enc);
+		msg->set_flags(IvpMsgFlags_RouteDSRC);
+		msg->addDsrcMetadata(tmx::messages::api::personalSafetyMessage_PSID);
+		msg->refresh_timestamp();
+
+		PLOG(logINFO) << " Pedestrian Plugin :: Broadcast PSM:: " << psmENC.get_payload_str();
+		auto *rMsg = dynamic_cast<routeable_message *>(msg.get());
+		BroadcastMessage(*rMsg);
+	}
+	else if(receivedType == "SensorDataSharingMessage")
+	{
+		sdsmMsg.set_contents(container.get_storage().get_tree());
+		sdsmENC.encode_j2735_message(sdsmMsg);
+
+		auto msg = std::make_unique<SdsmEncodedMessage>();
+		msg.reset();
+		msg.reset(dynamic_cast<SdsmEncodedMessage*>(factory.NewMessage(api::MSGSUBTYPE_SENSORDATASHARINGMESSAGE_STRING)));
+		std::string enc = sdsmENC.get_encoding();
+		msg->refresh_timestamp();
+		msg->set_payload(sdsmENC.get_payload_str());
+		msg->set_encoding(enc);
+		msg->set_flags(IvpMsgFlags_RouteDSRC);
+		msg->addDsrcMetadata(tmx::messages::api::sensorDataSharingMessage_PSID);
+		msg->refresh_timestamp();
+		
+		PLOG(logINFO) << " Pedestrian Plugin :: Broadcast SDSM:: " << sdsmENC.get_payload_str();
+		auto *rMsg = dynamic_cast<routeable_message *>(msg.get());
+		BroadcastMessage(*rMsg);
+	}
+	else if(receivedType == "TravelerInformation")
+	{
+		timMsg.set_contents(container.get_storage().get_tree());
+		timENC.encode_j2735_message(timMsg);
+
+		auto msg = std::make_unique<TimEncodedMessage>();
+		msg.reset();
+		msg.reset(dynamic_cast<TimEncodedMessage*>(factory.NewMessage(api::MSGSUBTYPE_TRAVELERINFORMATION_STRING)));
+		std::string enc = timENC.get_encoding();
+		msg->refresh_timestamp();
+		msg->set_payload(timENC.get_payload_str());
+		msg->set_encoding(enc);
+		msg->set_flags(IvpMsgFlags_RouteDSRC);
```

Comment:
Is this routed to RSU or just broadcast internally, then the Tim message is forwarded to ODE through ODEPlugin? 
![image](https://github.com/user-attachments/assets/2dcac318-64e9-4dc6-99be-0d550b992c28)

Reply:
It's broadcast internally then can be used by any plugin like ODE or Immediate Forward. I tested it with the Immediate Forward plugin and it worked fine